guides/how_sprockets_works.md

```diff
@@ -176,131 +181,152 @@ config.assets.precompile = [LOOSE_APP_ASSETS, /(?:\/|\\|\A)application\.(.css|js
 
-The third gem is the execjs gems. It allows you to run JavaScript code inside the Ruby environment and it uses the JavaScript environment that is available to you in the machine. We have some options that is working by default in the gem, like the Node.js environment and the V8 Google interpreter. And to use the gems is really simple. You actually run a JavaScript code inside the Ruby one.
+### Examples
 
 ```ruby
-require “execjs”
-require “open-uri”
-source = open(“http://coffeescript.org/extras/coffee-script.js”).read
+require 'execjs'
+ExecJS.eval "'red yellow blue'.split(' ')"
+# => ["red", "yellow", "blue"]
+```
+
+```ruby
+require 'execjs'
+require 'open-uri'
+source = open('https://coffeescript.org/v1/browser-compiler/coffee-script.js').read
 
 context = ExecJS.compile(source)
-context.call(“CoffeeScript.compiler”, “square = (x) -> x * x”, bare: true)
-# => “var square;\nsquare = function(x) {\n   return x * x;\n};”
+context.call('CoffeeScript.compile', 'square = (x) -> x * x', bare: true)
+# => "var square;\nsquare = function(x) {\n  return x * x;\n};"
 ```
 
 Here we are actually getting the CoffeeScript source code from the CoffeeScript website and compiling CoffeeScript code using Ruby. So as you can see, this gem is used by the coffee-script gem to compile CoffeeScript code to JavaScript. That brings us the the next gem, which is the coffee-rails gem.
 
 ## coffee-rails
 
-All this gem does is configures generators, so if you don't use generators, you actually don't need the gem. And it also defines a template handler so you can call handler CoffeeScript files from your controllers.
+All this gem does is configures generators, so if you don't use generators, you actually don't need the gem. It also defines a template handler so you can call handler CoffeeScript files from your controllers.
 
-### Assets generation in development
+### Asset generation in development
 
-In development, when you have this code, the javascript_include_tag,
+In development, when you are using the `javascript_include_tag` as below:
 
 ```erb
-<%= javascript_include_tag ‘application’%>
+<%= javascript_include_tag 'application'%>
 ```
-it's going to generate this HTML code that points to the digestive vessel of that file.
+
+The method generates the following HTML code that points to the digest version of that file:
 
 ```ruby
-javascript_include_tag ‘application’
-#<script
-#src=”/assets/application.debug-ddbd4593b22ac054471df143715c8ce65ef84938965c7db19d8a322950ec65b6.js”
-#></script>
+<script src="/assets/application.debug-ddbd4593b22ac054471df143715c8ce65ef84938965c7db19d8a322950ec65b6.js"></script>
+             | path
```

Comment:
I added this to try to better explain the segments of the asset path. It's not great, but I think it helps.